source/code/plugins/oms_common.rb

```diff
@@ -11,17 +11,494 @@ class Common
+      'Africa/Douala' => 'W. Central Africa Standard Time',
+      'Africa/Algiers' => 'W. Central Africa Standard Time',
+      'Africa/Libreville' => 'W. Central Africa Standard Time',
+      'Africa/Malabo' => 'W. Central Africa Standard Time',
+      'Africa/Niamey' => 'W. Central Africa Standard Time',
+      'Africa/Ndjamena' => 'W. Central Africa Standard Time',
+      'Africa/Tunis' => 'W. Central Africa Standard Time',
+      'Etc/GMT-1' => 'W. Central Africa Standard Time',
+      'Europe/Berlin' => 'W. Europe Standard Time',
+      'Europe/Andorra' => 'W. Europe Standard Time',
+      'Europe/Vienna' => 'W. Europe Standard Time',
+      'Europe/Zurich' => 'W. Europe Standard Time',
+      'Europe/Busingen' => 'W. Europe Standard Time',
+      'Europe/Gibraltar' => 'W. Europe Standard Time',
+      'Europe/Rome' => 'W. Europe Standard Time',
+      'Europe/Vaduz' => 'W. Europe Standard Time',
+      'Europe/Luxembourg' => 'W. Europe Standard Time',
+      'Europe/Monaco' => 'W. Europe Standard Time',
+      'Europe/Malta' => 'W. Europe Standard Time',
+      'Europe/Amsterdam' => 'W. Europe Standard Time',
+      'Europe/Oslo' => 'W. Europe Standard Time',
+      'Europe/Stockholm' => 'W. Europe Standard Time',
+      'Arctic/Longyearbyen' => 'W. Europe Standard Time',
+      'Europe/San_Marino' => 'W. Europe Standard Time',
+      'Europe/Vatican' => 'W. Europe Standard Time',
+      'Asia/Tashkent' => 'West Asia Standard Time',
+      'Antarctica/Mawson' => 'West Asia Standard Time',
+      'Asia/Oral' => 'West Asia Standard Time',
+      'Asia/Aqtau' => 'West Asia Standard Time',
+      'Asia/Aqtobe' => 'West Asia Standard Time',
+      'Indian/Maldives' => 'West Asia Standard Time',
+      'Indian/Kerguelen' => 'West Asia Standard Time',
+      'Asia/Dushanbe' => 'West Asia Standard Time',
+      'Asia/Ashgabat' => 'West Asia Standard Time',
+      'Asia/Samarkand' => 'West Asia Standard Time',
+      'Etc/GMT-5' => 'West Asia Standard Time',
+      'Pacific/Port_Moresby' => 'West Pacific Standard Time',
+      'Antarctica/DumontDUrville' => 'West Pacific Standard Time',
+      'Pacific/Truk' => 'West Pacific Standard Time',
+      'Pacific/Guam' => 'West Pacific Standard Time',
+      'Pacific/Saipan' => 'West Pacific Standard Time',
+      'Etc/GMT-10' => 'West Pacific Standard Time',
+      'Asia/Yakutsk' => 'Yakutsk Standard Time',
+      'Asia/Chita' => 'Yakutsk Standard Time',
+      'Asia/Khandyga' => 'Yakutsk Standard Time'
+    }
 
     class << self
-      
+      def get_current_timezone
+        return @@CurrentTimeZone if !@@CurrentTimeZone.nil?
+
+        tzID = 'Unknown'
+        
+        begin
+          md5sum = Digest::MD5.file('/etc/localtime').hexdigest
+          baseFolder = '/usr/share/zoneinfo/'
+          rightFolder = 'right/'
+          
+          Dir.glob("#{baseFolder}**/*") { |x|
+            if !File.directory? x and Digest::MD5.file(x).hexdigest == md5sum
```

Comment:
Use `File.file?` instead of `!File.directory?`</br>
Rename `x` to something meaningful.
This is calculating the hash of every file globbed but only enters the if clause when the md5 matches. Why not execute the if clause directly on /etc/localtime?